model/src/main/kotlin/Provenance.kt

```diff
@@ -66,11 +66,9 @@ data class RepositoryProvenance(
     val vcsInfo: VcsInfo,
 
     /**
-     * The original [VcsInfo] that was used to download the source code. It can be different to [vcsInfo] if any
-     * automatic detection took place. For example if the original [VcsInfo] does not contain any revision and the
-     * revision was automatically detected by searching for a tag that matches the version of the package there
-     * would be no way to match the package to the [Provenance] without downloading the source code and searching
-     * for the tag again.
+     * The original [VcsInfo] equal to the [meta-data of the package][Package.vcsProcessed], so with an empty resolved
+     * revision (as of 835b197). It is different from [vcsInfo] only in terms of the [resolved revision]
+     * [VcsInfo.resolvedRevision].
```

Comment:
> 2. All VcsInfo properties except for the resolved revision must be equal

Here, I'm having a problem with the argumentation. If the scan results says to have a scan result for id "foo" of (unresolved) revision "master", then we actually must never use than scan result, as we have no clue whether "master" changed since the time it was scanned.

In other words, shouldn't be exactly the opposite: We do not care about the `revision`s, but the `resolvedRevision`s must match in order to use a stored scan result.

What am I missing here?

Reply:
Commit message:

> does not involde

"does not involve"